keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ImageResolver.kt

```diff
@@ -32,32 +34,42 @@ class ImageResolver(
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
 
   // TODO: we may (probably will) want to make the parameter type more generic
-  suspend fun resolveImageId(resource: Resource<ServerGroupSpec>): Pair<String, String> =
-    when (val imageProvider = resource.spec.launchConfiguration.imageProvider) {
-      is IdImageProvider -> {
-        resolveFromImageId(imageProvider)
-      }
-      is ArtifactImageProvider -> {
-        resolveFromArtifact(resource, imageProvider)
-      }
-      is JenkinsImageProvider -> {
-        resolveFromJenkinsJob(resource, imageProvider)
-      }
-      else -> {
-        throw UnsupportedStrategy(imageProvider::class.simpleName.orEmpty(), ImageProvider::class.simpleName.orEmpty())
-      }
+  suspend fun resolveImageId(resource: Resource<ClusterSpec>): ResolvedImages =
```

Comment:
Note change of return type. This method can resolve the images for each region. I'm happier that there's a proper class for this as well.